skbio/stats/distance/_base.py

```diff
@@ -261,7 +261,7 @@ def from_iterable(
         iterable = list(iterable)
         keys_ = _get_keys(iterable, key, keys)
 
-        dm = np.empty((len(iterable),) * 2)
+        dm = np.zeros((len(iterable),) * 2)
```

Comment:
I am not sure if this is the right fix -- First, for `PairwiseMatrix`, all cell values are filled, therefore there is no need to pre-populate it with zero values. Pre-population is only needed for `SymmetricMatrix` with `validate=False`. Second, instead of pre-populating the entire matrix with 0, which is costly, you can only post-populate the diagonal line with 0 using `np.fill_diagonal`.